### PR TITLE
JOB/8138 Update the active support dependency

### PR DESCRIPTION
**Pull request by crpahl**
# What's in this PR?

Update the activesupport dependency to a version that is not affected by known security issues.

Github recommends upgrading to activesupport ~> 4.1.11

# QA
- [x] This has been QA'd by the PR owner
- [x] This has been QA'd by an approver

# References

- activesupport (4.0.2) suffers from https://nvd.nist.gov/vuln/detail/CVE-2015-3227

### CODE REVIEW

**Comment by crpahl** [comment]
@jessevanherk 

Just a few questions around testing if you don't mind! I'm not as familiar with Minitest and travis CI. Since we're updating the activesupport dependency to 4.2.9, I'm guessing there's no longer a need for the following files in the test/gemfiles folder:

```
Gemfile.rails-3.2.x
Gemfile.rails-4.0.x
Gemfile.rails-4.1.x
```

Since our dependency is getting updated to 4.2.x, I'm guessing there's no longer a need to test the previous versions? (obvious question, but I just want to make sure I'm not missing anything)

I'm thinking the .travis.yml file can also be updated from this:

```
sudo: false
language: ruby
script: bundle exec rake
rvm:
  - 1.9.3
  - 2.0.0
  - 2.1.5
  - 2.2.0
  - jruby-19mode
gemfile:
  - test/gemfiles/Gemfile.rails-3.2.x
  - test/gemfiles/Gemfile.rails-4.0.x
  - test/gemfiles/Gemfile.rails-4.1.x
  - test/gemfiles/Gemfile.rails-4.2.x
matrix:
  exclude:
    - rvm: 1.9.3
      gemfile: spec/gemfiles/rails-4.1.x.gemfile
    - rvm: 1.9.3
      gemfile: spec/gemfiles/rails-4.2.x.gemfile
  fast_finish: true
```

To this:

```
sudo: false
language: ruby
script: bundle exec rake
rvm:
  - 1.9.3
  - 2.0.0
  - 2.1.5
  - 2.2.0
  - jruby-19mode
gemfile:
  - test/gemfiles/Gemfile.rails-4.2.x
matrix:
  exclude:
    - rvm: 1.9.3
      gemfile: spec/gemfiles/rails-4.2.x.gemfile
  fast_finish: true
```

^ Basically remove anything that isn't rails 4.2?

And finally, I'm not sure why this line was included in the Gemfile:

```
gem "activesupport", "4.0.2"
```

But I feel like that can be removed as well possibly (since the required dependency is in the gemspec file)?

**Comment by jessevanherk** [comment]
> Gemfile.rails-3.2.x
> Gemfile.rails-4.0.x
> Gemfile.rails-4.1.x
> Since our dependency is getting updated to 4.2.x, I'm guessing there's no longer a need to test the previous versions? (obvious question, but I just want to make sure I'm not missing anything)

> Basically remove anything that isn't rails 4.2?

I agree. There's no value in supporting old/end-of-life rails 3 or rails4 versions, so makes sense to remove these from each of the spots they're being used.

However, since hashaccessor is a "public" gem, there's some need to test on multiple versions of rails.  If anything we should be making sure it works with newer rails. That's out-of-scope for this PR though - would be good to have a followup story to make this gem compatible with rails5 and rails6.

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.1.5
+2.3.4
```
Comment (crpahl) [inline .ruby-version]: I updated the rvm version to match our version as well since it was lagging pretty far behind.

**Comment by linds14sr20det** [comment]
LGTM.

**Pull request merged**